### DISCUSSION TITLE
[Chapter 09] 기술 중심 패키징에 대한 고민

### DISCUSSION THREAD

**Discussion by coalong** [cowaine/is-my-code-that-strange#83] [Chapter 09-10 (악마, 이름 설계)] [answered]
저는 책에 있는 예제를 작성할 때나 사이드 플젝을 할 때 패키징을 나누고 클래스를 작성하는 것이 늘 고민되고 시간이 오래 걸리는 것 같습니다.

소규모 플젝에서는 MVC 패턴에 맞게 폴더 구성이 될 수 있지만, 규모가 커지면 관리하고 유지보수하기 어려울 것 같습니다.

p.204-205 에서는 패키징에 따라 악마가 생길 수 있다고 설명하고있고 p.206에서는 비즈니스 클래스에 따라 패키징을 분리한 예시를 설명하며 권장합니다. 
실제 업무에서 이렇게 비즈니스 클래스에 따라 패키징할 수 있을까요? (재고, 주문, 결제 등 서로 참조할 상황? 일이 없는 지 궁금)

여러분들은 회사나 플젝에서 어떤 구조로 패키징이 되어있고, 어떤 패키징 방법을 선호하는지 궁금합니다.

**Comment by snoopingh** [comment, marked as answer]
어떤 방식을 딱 정해서 선호하지는 않습니다만, 지금 회사에서 운영중인 서비스는 아주 명확하게 기술중심패키징으로 되어 있네요.

실제 업무에서 이렇게 비즈니스 클래스에 따라 패키징할 수 있을까요? (재고, 주문, 결제 등 서로 참조할 상황? 일이 없는 지 궁금)
> 위 질문과 관련해서는 DDD에서 나오는 개념인 애그리거트를 한번 찾아보시면 좋을 것 같습니다.
   책의 예시는 너무 간단명료해서 저도 좀 이해가 안 가는 부분이 있네요
   결국 저자가 악마라고 일컫는 부분들을 해소하기 위해 진화한 아키텍처가 도메인 중심의 개발 같습니다.

**Comment by firewoody237** [comment]
최근 도메인주도설계 기본 책을 읽고 있는데,

값객체>도메인>도메인서비스>애플리케이션서비스 순으로 규모가 커지는데, 말씀해주신 패키지간 커뮤니케이션은 애플리케이션 서비스의 역할이 아닐까 싶습니다.
저희 회사는 MSA로 서비스를 각자 떼가고 있어서, 점점 더 플랫폼들이 MSA화 되면 필요한 API만 서로 건네기에 좀 더 도메인 중심이 되지 않을까 싶네요.

(아직 도메인주도설계 책을 다 안읽어서 잘못된 정보일 수도 있습니다 ㅎㅎ - 스누피님이 얘기하신 애그리거트는 아직 진도를 못나갔어요)
